### DISCUSSION TITLE
Teams functionalities - possible improvements

### DISCUSSION THREAD

**Discussion by cix99** [spatie/laravel-permission#1905] [Ideas]
Hi everyone.

Is is great to know that the package now supports teams based roles/permissions.
The package itself is great and now it might fit for most multi-tenant apps too.

I think there's only few steps to improve the teams functionality, and these are:
- Possibility to retrieve the Teams that a User belongs to, based on roles/permissions assigned
- Possibility to retrieve Users that belong to a Team, based on roles/permissions assigned.

These improvements would be very useful for example when creating a teams page, where the user can choose a Team to manage. As it's structured now, the getAllPermissions() first requires the app to pass the team id into the setPermissionsTeamId(), and then it gets all the permissions only for that team id. Would be great to get all the permissions instead.

As Teams is a new feature, I think that we could put in place some ideas to improve this package even more.

**Comment by erikn69** [comment]
Also there is  [spatie/laravel-multitenancy](https://github.com/spatie/laravel-multitenancy)

>Possibility to retrieve the Teams that a User belongs to, based on roles/permissions assigned

As team model is defined from the custom app, not from this package, it's not posible add relations or scopes for that
Look at the picture on https://github.com/spatie/laravel-permission/pull/1804, it is easy to do a relation on teams and user

>Possibility to retrieve Users that belong to a Team, based on roles/permissions assigned.

#1910

------
you can make a pull request because maybe no one else will

**Reply by erikn69** [reply]
@cix99 sorry, i tried, https://github.com/spatie/laravel-permission/pull/1910#issuecomment-960653613
>To keep the package simple, I'd like to not pull in any new scope for now.
